src/Encoders/OffloadEncoderEncrypted.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Aol\Offload\Encoders;
+
+use Aol\Offload\Exceptions\OffloadConfigurationException;
+use Aol\Offload\Exceptions\OffloadEncoderException;
+
+abstract class OffloadEncoderEncrypted implements OffloadEncoderInterface
+{
+    /** @var OffloadEncoderInterface The base encoder. */
+    private $base;
+    /** @var string A key ID present in the keys to use as the main key. */
+    private $encryption_key_id;
+    /** @var string[] An associative array of key ID to key. */
+    private $keys;
+    /** @var bool Whether to use the base encoder on a decode error. */
+    private $use_base_on_decode_error;
+
+    /**
+     * Create a new encrypted encoder.
+     *
+     * @param OffloadEncoderInterface $base                     The encoder to encrypt.
+     * @param string                  $encryption_key_id        A key ID present in the keys to use as the main key.
+     * @param string[]                $keys                     An associative array of key ID to key.
+     * @param bool                    $use_base_on_decode_error Whether to use the base encoder on a decode error.
+     *
+     * @throws OffloadConfigurationException If there was an issue with the options.
+     */
+    public function __construct(
+        OffloadEncoderInterface $base,
+        $encryption_key_id,
+        array $keys,
+        $use_base_on_decode_error = false
```

Comment:
Why not make this an instance of `OffloadEncoderInterface` and let the end developer choose what the base is?